pkg/kubelet/volumes.go

```diff
@@ -143,7 +143,8 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 			return nil, err
 		}
 		if attacher != nil {
-			err = attacher.Attach(volSpec, kl.hostname)
+			host := &volumeHost{kl}
+			err = attacher.Attach(host, volSpec, kl.hostname)
```

Comment:
aws and gce `AttachDisk` return different results: aws returns device path and error, while gce just returns error. It will be great if `attacher.Attach` returns the union of both cases. Otherwise, the aws has to remember the attached device path.

Same to `DetachDisk`